crossfit/backend/gpu.py

```diff
@@ -30,7 +33,14 @@ def _get_gpu_count():
     # that are incompatible with Dask-CUDA. If CUDA runtime functions are
     # called before Dask-CUDA can spawn worker processes
     # then Dask-CUDA it will not work correctly (raises an exception)
-    nvml_device_count = nvml.device_get_count()
+    if device_get_count is not None:
+        nvml_device_count = device_get_count()
+    else:
+        try:
+            nvml_device_count = pynvml.nvmlDeviceGetCount()
+        except Exception:
+            nvml_device_count = 0
```

Comment:
Would love an opinion here. This code works as is, but I am not sure if the logic is sound enough to cover all cases here.

Was wondering if I should consider copying over relevant functions from https://github.com/dask/distributed/blob/main/distributed/diagnostics/nvml.py ?